src/commands/traces/drivers/byovd.rs

```diff
@@ -2,76 +2,147 @@
+
+        let is_administrator = match is_administrator() {
+            Ok(is_administrator) => is_administrator,
+            Err(error) => {
+                displayer.failure(
+                    format!(
+                        "Failed to check administrator privileges: {}",
+                        error.message()
+                    )
+                    .as_str(),
+                );
+
+                return ExitCode::FAILURE;
+            }
+        };
+
+        if !is_administrator {
+            displayer.failure("The application needs administrator privileges to run the driver");
+
+            return ExitCode::FAILURE;
         }
 
-        if !self.path.try_exists()? || !self.path.is_file() {
-            return Ok(());
+        displayer.success("The application has administrator privileges");
+        displayer.loading("Checking the driver path validity");
+
+        let driver_exists = match self.driver.try_exists() {
+            Ok(exists) => exists,
+            Err(error) => {
+                displayer.failure(
+                    format!("Failed to check the existence of the driver path: {error}").as_str(),
+                );
+
+                return ExitCode::FAILURE;
+            }
+        };
+
+        if !driver_exists || !self.driver.is_file() {
+            displayer.failure("The driver path does not exist or is not a file path");
+
+            return ExitCode::FAILURE;
         }
 
-        unsafe {
-            let service_manager: Owned<SC_HANDLE> = Owned::new(OpenSCManagerW(
-                PCWSTR::null(),
-                PCWSTR::null(),
-                SC_MANAGER_CREATE_SERVICE,
-            )?);
+        displayer.success("The driver path is valid");
+        displayer.loading("Creating the service manager");
 
-            if OpenServiceW(
-                *service_manager,
-                &HSTRING::from(self.service_name.as_str()),
-                GENERIC_READ.0,
-            )
-            .is_ok()
-            {
-                return Ok(());
+        let service_manager: Owned<SC_HANDLE> = match unsafe {
+            OpenSCManagerW(PCWSTR::null(), PCWSTR::null(), SC_MANAGER_CREATE_SERVICE)
+        } {
+            Ok(handle) => unsafe { Owned::new(handle) },
+            Err(error) => {
+                displayer.failure(
+                    format!("Failed to create the service manager: {}", error.message()).as_str(),
+                );
+
+                return ExitCode::FAILURE;
             }
+        };
 
-            let service: Owned<SC_HANDLE> = Owned::new(CreateServiceW(
+        displayer.success("The service manager is created");
+        displayer.loading("Creating the driver service");
+
+        let service: Owned<SC_HANDLE> = match unsafe {
+            CreateServiceW(
                 *service_manager,
                 &HSTRING::from(self.service_name.as_str()),
                 &HSTRING::from(self.displayed_name.as_str()),
```

Comment:
![image](https://github.com/user-attachments/assets/c3220ed7-fe42-450b-bf11-3ad195be9f35)
Get a bug `:` is added at the end

Reply:
Close : Autorun Display not a rust bug